### PR TITLE
Add try/catch arounds reset timeout so we dont blow up on shutdown

### PR DESCRIPTION
**Pull request by hamnis**
Can be enhanced by including: https://github.com/http4s/blaze/pull/535

### CODE REVIEW

File: blaze-core/src/main/scala/org/http4s/blazecore/IdleTimeoutStage.scala [hunk 1]
```diff
@@ -83,7 +84,7 @@ final private[http4s] class IdleTimeoutStage[A](
       case r: RuntimeException if r.getMessage == "TickWheelExecutor is shutdown" =>
```
Comment (rossabaker) [inline blaze-core/src/main/scala/org/http4s/blazecore/IdleTimeoutStage.scala:84]: When blaze-0.14.7 is out, we can do better here with the new Exception type.

File: blaze-core/src/main/scala/org/http4s/blazecore/IdleTimeoutStage.scala [hunk 2]
```diff
@@ -79,13 +79,15 @@ final private[http4s] class IdleTimeoutStage[A](
...
-      case r: RuntimeException if r.getMessage == "TickWheelExecutor is shutdown" =>
-        logger.warn(s"Resetting timeout after tickwheelexecutor is shutdown")
-        cancelTimeout()
-      case NonFatal(e) => throw e
-    }
+    if (exec.isAlive) {
+      try setAndCancel(exec.schedule(killSwitch, ec, timeout))
+      catch {
+        case TickWheelExecutor.AlreadyShutdownException =>
+          logger.warn(s"Resetting timeout after tickwheelexecutor is shutdown")
```
Comment (hamnis) [inline blaze-core/src/main/scala/org/http4s/blazecore/IdleTimeoutStage.scala:86]: drop log?

Comment (rossabaker) [reply blaze-core/src/main/scala/org/http4s/blazecore/IdleTimeoutStage.scala:86]: How frequent is the error still? It still indicates something is broken. It seems like a good warning to me.

**Review by ChristopherDavenport — APPROVED** [review]
Can always lower log level/remove if its too loud/noisy.

**Pull request merged**